### ISSUE TITLE
Example Application

### ISSUE THREAD

**Issue by JakeAustwick** [dobtco/formbuilder-rb#5]
I think it would be useful to get an example app up actually integrated with the JS library, I'm having a hard time grasping some things and figure that a basic example would help.

**Comment by ajb** [comment]
FWIW, I think this is a pretty good idea. My main struggle right now is that I'm not really sure this gem is useful to anyone but our organization, since it's forces a pretty specific usage pattern. 

Did you figure this out?

**Comment by JakeAustwick** [comment]
Yeah I figured it out, however I had to hack at the gem quite a lot for my specific needs because it did seem as though it was made for internal usage at certain points.

For example a draft_only hidden parameter, and the submit button was hardcoded to 'Save Draft' or something along them lines, + a few other little things along the way (such as disabling autosave). I'm using it to provide a formbuilder type solution for my users, similar to that of what wufoo provides.

I'd commit my changes back but I'm really just a backend coder, I ended up just hacking away at the final .js file, because I don't know coffeescript either.

I also have a question or two, I guess I may just ask them here as they aren't really worth opening an issue for, just a few quick queries.

What is the responses_text field for on the entry model? It just joins the fields/entries by a space. This seems entirely useless? You couldn't even split(' ') because the field entries themselves can contain spaces.

**Comment by ajb** [comment]
Yeah, I'm in the progress of removing the autosave stuff from this gem, although I'm also going to be introducing a dependency on [erector](http://www.github.com/adamjacobbecker/erector-rails4) for the views.

I'm using the `responses_text` to provide a search index -- just collecting everything there, and then converting it to a tsvector in a Postgres trigger.

Anyway, I'm glad you at least found this somewhat useful. Cheers!

**Comment by JakeAustwick** [comment]
Have you considered basic conditional logic for the forms?

It's something I'm going to have to implement myself, and just wondered whether you had considered it.

I'm talking only really basic stuff, like:
- Only show this field if X is checked
- Only show this field if X Dropdown == Y

And I found it more than somewhat useful, it's saved me a reasonable amount of time.

**Comment by ajb** [comment]
It's not something I've thought about, but I agree that it would be a great addition. If I can help you at all w/ implementing it, let me know. I'm assuming you've taken a look at [Formbuilder.js](https://github.com/dobtco/formbuilder) -- it would be great to have in there too.

**Comment by JakeAustwick** [comment]
Yeah, I'm using this with Formbuilder.js.

I'm going to have to dive into the source of the JS a little deeper before I take this on, because my JS knowledge is very limited.

However from a Rails perspective, I see this being an easy addition.

I guess I see two options in terms of the submitting entries on a conditional form:
- In the JS, upon the form save, don't POST the fields which are currently hidden due to logic to the server. (remove the inputs essentially)
- Post all the form fields as per usual, even if they are hidden due to conditional logic, and then just remove them at the rails end before validation occurs and they are saved in the DB.

Do you have a preference for either, or can you see any clear winner?

**Comment by ajb** [comment]
I would choose the first option -- it's pretty easy when hiding the field using jquery (or whatever) to also disable it, a la:

``` javascript
$('#field_wrapper').hide().find(':input').prop('disabled', true)
```

**Comment by JakeAustwick** [comment]
Alright, makes sense, I have zero backbone experience either, probably not going to help ;). I'll probably give this a go at some point soon, although don't your your breath. My chances of succeeding are slim.

**Comment by JakeAustwick** [comment]
Just for reference, this is the kind of thing I'm trying to achieve:
http://i.imgur.com/6qSy9cj.png

**Comment by jcutrell** [comment]
I've actually been working on a fork, and building out a test application of sorts.

A few things that probably need to be covered:
- How to extend Formbuilder.js without breaking STI (was stuck on this for a bit)
- How to extend Formbuilder.js existing events (with things like "update") - this seems like a pretty common necessity, to do things like masking or creating nicer dropdowns with something like Chosen.js
- Easily switch between Carrierwave and Paperclip (the differences are very easy to accomodate for)
- Actually render a form entry. Looks like this isn't currently implemented in the dummy application.

Would love to work on this if we can get a feature set defined, maybe in a different issue/milestone?
